python/lsst/pipe/tasks/diffractionSpikeMask.py

```diff
@@ -0,0 +1,436 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+__all__ = ["DiffractionSpikeMaskConfig", "DiffractionSpikeMaskTask"]
+
+
+import math
+import numpy as np
+import astropy.units as u
+
+import lsst.afw.geom as afwGeom
+from lsst.afw.image import abMagErrFromFluxErr
+import lsst.geom
+from lsst.pex.config import Config, ConfigField, Field
+from lsst.pipe.base import Struct, Task
+from lsst.meas.algorithms import getRefFluxField, LoadReferenceObjectsConfig
+import lsst.sphgeom
+from .colorterms import ColortermLibrary
+
+
+class DiffractionSpikeMaskConfig(Config):
+    """Config for BrightStarMaskTask.
+    """
+    refObjLoader = ConfigField(dtype=LoadReferenceObjectsConfig,
+                               doc="Configuration of reference object loader")
+    applyColorTerms = Field(
+        dtype=bool,
+        default=False,
+        doc=("Apply photometric color terms to reference stars?\n"
+             "`True`: attempt to apply color terms; fail if color term data is "
+             "not available for the specified reference catalog and filter.\n"
+             "`False`: do not apply color terms."),
+        optional=True,
+    )
+    colorterms = ConfigField(
+        dtype=ColortermLibrary,
+        doc="Library of photometric reference catalog name: color term dict"
+        " (see also applyColorTerms).",
+    )
+    photoCatName = Field(
+        dtype=str,
+        optional=True,
+        doc=("Name of photometric reference catalog; used to select a color"
+             " term dict in colorterms. See also applyColorTerms."),
+    )
+    raKey = Field(
+        dtype=str,
+        default="coord_ra",
+        doc="RA column name in the reference catalog.",
+    )
+    decKey = Field(
+        dtype=str,
+        default="coord_dec",
+        doc="Declination column name in the reference catalog.",
+    )
+    angleMargin = Field(
+        dtype=float,
+        default=60.,
+        doc="Margin outside the exposure bounding box to include bright "
+        "sources. In arcseconds.",
+    )
+    magnitudeThreshold = Field(
+        dtype=float,
+        default=15,
+        doc="Threshold magnitude for treating a star from the reference catalog"
+        " as bright.",
+    )
+    diffractAngle = Field(
+        dtype=float,
+        default=45,
+        doc="Angle in degrees of the location of diffraction spikes with "
+        "respect to camera at 0 rotation angle.",
+    )
+    spikeAspectRatio = Field(
+        dtype=float,
+        default=10,
+        doc="Ratio of the length of a diffraction spike to it's width in the"
+        " core of the star.",
+    )
```

Comment:
Do any of these vary per band?  Is that something that may be added in the future?

Reply:
Bruno's analysis did not show a dependence on band, so we decided to keep this simple for now. I would like to update the calculations to be based on counts in the future, so that it would properly scale with exposure time.